plugins/main/public/components/common/welcome/agents-welcome.js

```diff
@@ -227,16 +229,14 @@ export const AgentsWelcome = compose(
         welcome: 8,
       });
       const filterHandler = new FilterHandler(AppState.getCurrentPattern());
-      const $injector = getAngularModule().$injector;
+
       this.drawerLokedSubscribtion = getChrome()
         .getIsNavDrawerLocked$()
         .subscribe(isLocked => {
           this.setState({ isLocked }, () => {
             this.updateWidth();
           });
         });
-      this.router = $injector.get('$route');
-      this.location = $injector.get('$location');
       window.addEventListener('resize', this.updateWidth); //eslint-disable-line
       await VisFactoryHandler.buildAgentsVisualizations(
```

Comment:
Is this still necessary?

Reply:
Yes, it's necessary to update the url when the user unpinned the agent